tests/test_spark.py

```diff
@@ -0,0 +1,123 @@
+#
+# Copyright (c) Granulate. All rights reserved.
+# Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
+#
+
+import logging
+from time import sleep
+from typing import List
+
+import pytest
+from docker import DockerClient
+from docker.models.containers import Container
+from docker.types import Mount
+from granulate_utils.metrics import MetricsSnapshot
+from granulate_utils.metrics.metrics import (
+    SPARK_AGGREGATED_STAGE_METRICS,
+    SPARK_APPLICATION_DIFF_METRICS,
+    SPARK_APPLICATION_GAUGE_METRICS,
+    SPARK_EXECUTORS_METRICS,
+)
+from granulate_utils.metrics.sampler import BigDataSampler
+from pytest import LogCaptureFixture
+
+from gprofiler.log import get_logger_adapter
+from tests.conftest import _build_image
+from tests.utils import _wait_container_to_start
+
+# `BigDataSampler` receives a logger as an argument.
+logger = get_logger_adapter("gprofiler_spark_test")
+
+# List that includes all the metrics that are expected to be collected by `BigDataSampler` in SA mode.
+EXPECTED_SA_METRICS_KEYS = [
+    metric
+    for metrics_dict in (
+        SPARK_APPLICATION_GAUGE_METRICS,
+        SPARK_APPLICATION_DIFF_METRICS,
+        SPARK_AGGREGATED_STAGE_METRICS,
+        SPARK_EXECUTORS_METRICS,
+    )
+    for metric in metrics_dict.values()
+]
+
+SPARK_MASTER_HOST = "127.0.0.1"
+
+DISCOVER_INTERVAL_SECS = 5
+DISCOVER_TIMEOUT_SECS = 60
+
+
+@pytest.fixture(scope="function")
```

Comment:
function scope is default, no need to specify it explicitly

Reply:
b29f95b